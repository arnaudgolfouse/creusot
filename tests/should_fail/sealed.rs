extern crate creusot_contracts;
use creusot_contracts::*;

pub trait Tr {
    #[logic]
    fn f(self, x: Int) -> Int;

    #[logic(sealed)]
    #[open]
    fn g(self, x: Int) -> Int {
        self.f(x) + 1
    }
}

impl Tr for Int {
    #[logic]
    fn f(self, x: Int) -> Int {
        self + x
    }

    #[logic]
    fn g(self, x: Int) -> Int {
        x
    }
}
