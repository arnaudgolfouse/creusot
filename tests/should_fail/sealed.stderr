error: This trait methods overrides a sealed implementation.
  --> sealed.rs:22:5
   |
22 |     fn g(self, x: Int) -> Int {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: This sealed implementation is overriden.
  --> sealed.rs:10:5
   |
10 |     fn g(self, x: Int) -> Int {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

